homeassistant/components/obihai/button.py

```diff
@@ -32,28 +19,25 @@ async def async_setup_entry(
     async_add_entities: entity_platform.AddEntitiesCallback,
 ) -> None:
     """Set up the Obihai sensor entries."""
-    username = entry.data[CONF_USERNAME]
-    password = entry.data[CONF_PASSWORD]
-    host = entry.data[CONF_HOST]
-    requester = ObihaiConnection(host, username, password)
+    requester = hass.data[DOMAIN][entry.entry_id]["requester"]
 
     await hass.async_add_executor_job(requester.update)
-    buttons = [ObihaiButton(requester.pyobihai, requester.serial)]
+    buttons = [ObihaiButton(requester.pyobihai, requester.device_info, "Reboot")]
     async_add_entities(buttons, update_before_add=True)
 
 
-class ObihaiButton(ButtonEntity):
+class ObihaiButton(ObihaiEntity, ButtonEntity):
     """Obihai Reboot button."""
 
-    entity_description = BUTTON_DESCRIPTION
-
-    def __init__(self, pyobihai: PyObihai, serial: str) -> None:
-        """Initialize monitor sensor."""
-        self._pyobihai = pyobihai
-        self._attr_unique_id = f"{serial}-reboot"
+    device_class = ButtonDeviceClass.RESTART
 
     def press(self) -> None:
         """Press button."""
 
         if not self._pyobihai.call_reboot():
             raise HomeAssistantError("Reboot failed!")
+
+    @property
+    def entity_category(self):
+        """Reboot is a configuration entity."""
+        return EntityCategory.CONFIG
```

Comment:
This is static, this can be assigned as a class attribute (next to the above device class):

```py
_attr_entity_category = EntityCategory.CONFIG
```